crates/services/src/seqlock.rs

```diff
@@ -0,0 +1,132 @@
+//! A simple implementation of a sequential lock.
+//! More details: <https://docs.kernel.org/locking/seqlock.html>
+//! Optimized for occasional writes and frequent reads
+//! !!WARNING!!
+//! ONLY USE IF ALL THE BELOW ARE MET
+//! 1. Internal data < 64 bytes
+//! 2. Data is Copy
+//! 3. ONLY 1 writer
+//! 4. VERY frequent reads
+
+use std::{
+    cell::UnsafeCell,
+    sync::atomic::{
+        AtomicU64,
+        Ordering,
+    },
+};
+
+/// A simple implementation of a sequential lock.
+/// some usage of unsafe, T must be Copy
+#[derive(Debug)]
+pub struct SeqLock<T> {
+    sequence: AtomicU64,
+    data: UnsafeCell<T>,
```

Comment:
We could create a trait for this `T` to guarantee that it's below 64 bytes and provide some default impls for the types we expect to see.

```rs
/// Type is equal to or smaller than 64 bytes
pub trait Small;
```

Reply:
(chiming in) perhaps `Small` should imply `Copy` in this case: `pub trait Small: Copy;`

Reply:
i don't quite like the idea of creating default impls here ~ we won't be able to implement the trait for types outside the crate either due to the orphan rule.

I can enforce the type size at runtime if you'd like.